contributor-docs/adrs/adr-008-experimental-components.md

```diff
@@ -0,0 +1,77 @@
+
+The approach that has served us well since Dec 2021 has been "drafts" (along with causing a lot of confusion).
+
+[ADR-006](./adr-006-drafts.md) explains our use of drafts, but here are the relevant parts:
+
+- When we are not sure about the API of a component, we put it in a different bundle: "primer/react/drafts"
+- A draft has experimental API, but follows the development conventions of primer/react
+- Components in drafts are accompanied by a design spec and go through the process of accessibility reviews. A component can check multiple boxes in the component lifecycle.
+- The contract with consumers is that you are opting into a component in which we don't have 100% confidence yet.
+
+[^1]: Calling this a prototype and not experimental, because that already means something else in the maturity lifecycle.
+
+&nbsp;
+
+#### Problems with drafts
+
+1. **Incompatible with semver**: Because the expectation with drafts is that we are not 100% confident about the API or structure of a component, we slip in breaking changes to draft components as minor releases to primer/react. On the consumer side, this makes every upgrade, even a patch, a potentially breaking upgrade. So, you are effectively opting out of your semver.
+
+   Upstreaming a component that was built in the context of a product requires re-evaluating the baked-in assumptions and most likely a change in the API to make it useful for other contexts.
+
+1. **Confusion about maturity**: Because of the generic name, "drafts" causes a lot of confusion, especially with "experimental". It is a prototype or experiment, but NOT in the way the [component lifecycle]([https://primer.style/contribute/component-lifecycle]) defines "Experimental" maturity.
+
+1. **Need for experimental candidates**: As more teams build React components and look to upstream reusable components, it is natural to look at drafts as the place to put them. However, the barrier to entry into drafts isn't low (because we still follow design reviews, dev conventions, API reviews, etc.)
+
+   Feature teams using Primer may not be intimately familiar with Primer component lifecycle criteria or have the time and/or knowledge to meet it. Often, feature teams create their own components outside of Primer if they find it too difficult to contribute. This leads to lower Primer adoption and conflicting versions of similar functionality throughout GitHub products.
+
+   We want to enable feature teams to contribute to Primer. We also want to make it clear which contributions haven't been evaluated by our high standards for Primer components.
+
+
+
+&nbsp;
+
+## Decision
+
+1. npm package: "Experimental" components, as defined by the component lifecycle, should not be part of semantically versioned npm package for `@primer/react`.
+
+2. npm package: Each experimental component should be independently published and versioned with semver. Breaking changes should be tagged as major versions. Example: `@github/react-commentbox`
```

Comment:
I would question wanting to dump more in the `@github` scope. We might want to make a private new team to house and manage these components. `@drywall`? Since it comes before Primer? 😛 Either way, we'll be publishing a number of components I'm sure, so scoping them would be wise, but not under `@github` or `@primer` IMHO

Reply:
Love drywall! 🤣 

> I would question wanting to dump more in the @github scope. 

I don't think I understand the tradeoff here, what's the reason for not publishing under github scope?

Reply:
> I don't think I understand the tradeoff here, what's the reason for not publishing under github scope?

I have the same question.

Reply:
by publishing under the `@github` scope, especially on a per-component basis, this will flood the scope with repos and namespaces. Since there will be a number of components, it would (IMHO) make sense to be bundled up into its own org for cleanliness.